### ISSUE TITLE
Crash in ScopedTimeCache::Now before main function started

### ISSUE THREAD

**Issue by void-mian** [grpc/grpc#38167]
### What version of gRPC and what language are you using?
gRPC 1.60.0, c++

### What operating system (Linux, Windows,...) and version?
Linux(ubuntu24.04.1LTS)

### What runtime / compiler are you using (e.g. python version or version of gcc)
clang18.1.3

### What did you do?
minimal reproduction:
```cpp
#include <grpcpp/alarm.h>

grpc::Alarm not_used;

int main() {
    return 0;
}
```

### What did you expect to see?
no error and SIGSEGV

### What did you see instead?
Got SIGSEGV in line `previous()->InvalidateCache()` (function: ScopedTimeCache::Now, file: vcpkg/buildtrees/grpc/src/v1.60.0-2dcaca28c0.clean/src/core/lib/gprpp/time.cc), the `previous()` is `NoDestructSingleton<GprNowTimeSource>::Get()` a few line above and `NoDestructSingleton` make it zero init, then I got SIGSEGV while calling `InvalidateCache()`.

![image](https://github.com/user-attachments/assets/a6797485-cc48-4ef4-a41a-1af4f1afac7b)
![image](https://github.com/user-attachments/assets/5e185a79-9735-4c90-a51b-ac91c1208c34)

**Comment by yashykt** [comment]
I don't think we support `grpc::Alarm` or any of the other grpc types as global variables. Please try moving it to the stack.
